src/fluree/db/api.cljc

```diff
@@ -806,3 +806,34 @@
   ([db opts]
    (let [grouping (:group-by opts)]
      (reasoner/reasoned-facts db grouping))))
+
+(defn trigger-index
+  "Manually triggers indexing for a ledger without requiring a transaction.
+  
+  This is useful for external indexing processes (e.g., AWS Lambda) that need
+  to ensure a ledger is indexed without creating new transactions.
+  
+  Parameters:
+    conn - Database connection
+    ledger-alias - The alias/name of the ledger to index
+    opts - (optional) Options map:
+      :branch - Branch name (defaults to main branch)
+      :block? - If true, waits for indexing to complete before returning
```

Comment:
I think having boolean options for control flow increases code complexity. I think things would be simpler with two different functions in each place where the `:block?` option is read instead.